packages/worker/src/worker.ts

```diff
@@ -1498,8 +1498,11 @@ export class Worker {
   }
 
   protected activity$(): Observable<void> {
-    // Note that we poll on activities even if there are no activities registered.
-    // This is so workflows invoking activities on this task queue get a non-retryable error.
+    // This Worker did not register any activities, return early
+    if (this.workflowCreator === undefined) {
+      this.activityPollerStateSubject.next('SHUTDOWN');
+      return EMPTY;
+    }
```

Comment:
You should log a warning in this situation. We might somewhere else already I think, but users can easily get confused.

Reply:
Yes, that's a good idea